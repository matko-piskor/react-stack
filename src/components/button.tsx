import { type HTMLProps } from 'react';

type Props = Omit<HTMLProps<HTMLButtonElement>, 'type'> & { type?: ButtonType; 'data-testid'?: string };

type ButtonType = 'button' | 'submit' | 'reset' | undefined;

export function Button({ children, ...props }: Props) {
    return (
        <button className='bg-red-900 px-8 py-4 text-white' {...props}>
            {children}
        </button>
    );
}
