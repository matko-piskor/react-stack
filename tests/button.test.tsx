import { Button } from '../src/components/button';
import { render, screen } from '@testing-library/react';

test('Button should render', () => {
    render(<Button>Test</Button>);

    const btn = screen.getByRole('button');

    expect(btn).toBeInTheDocument();
});

test('Button should match spanshot', () => {
    const { asFragment } = render(<Button>Test</Button>);

    expect(asFragment()).toMatchSnapshot();
});

test('Button should call onClick', () => {
    const onClick = vi.fn();
    render(<Button onClick={onClick}>Test</Button>);

    const btn = screen.getByRole('button');
    btn.click();

    expect(onClick).toHaveBeenCalled();
    expect(onClick).toHaveBeenCalledTimes(1);
});
