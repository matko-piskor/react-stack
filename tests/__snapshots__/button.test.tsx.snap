// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button should match spanshot 1`] = `
<DocumentFragment>
  <button
    class="bg-red-900 px-8 py-4 text-white"
  >
    Test
  </button>
</DocumentFragment>
`;
